health_monitor/src/health_monitor.cpp

```diff
@@ -22,38 +22,52 @@
 namespace health_monitor
 {
 
+
+    HealthMonitor::HealthMonitor()
+    {
+        std::vector<bool>pub_truck(8, false);
+        std::vector<bool>pub_car(6, false);
```

Comment:
These are local variables since you have declared them here.
The correct syntax would be
```
 pub_truck = std::vector<bool>(8, false)
```